synthesis/codon/codon.go

```diff
@@ -349,16 +418,48 @@ func generateCodonTable(aminoAcids, starts string) codonTable {
 	for k, v := range aminoAcidMap {
 		aminoAcidSlice = append(aminoAcidSlice, AminoAcid{string(k), v})
 	}
-	return codonTable{startCodons, stopCodons, aminoAcidSlice}
+
+	// generate a map of codons -> amino acid
+
+	var translationMap = make(map[string]string)
+	for _, aminoAcid := range aminoAcidSlice {
+		for _, codon := range aminoAcid.Codons {
+			translationMap[codon.Triplet] = aminoAcid.Letter
+		}
+	}
+
+	// GenerateStartCodonTable returns a mapping from the start codons of a Table to their associated amino acids.
+	// For our codonTable implementation, assumes that we always map to Met.
+
+	startCodonsMap := make(map[string]string)
+	for _, codon := range startCodons {
+		startCodonsMap[codon] = "M"
+	}
+
+	// This function is run at buildtime and failure here means we have an invalid codon table.
+	chooser, err := newAminoAcidChoosers(aminoAcidSlice)
+	if err != nil {
+		panic(fmt.Errorf("tried to generate an invalid codon table %w", err))
```

Comment:
not sure if this should panic and is also not covered by tests.

Reply:
`generateCodonTable` runs at build time when the hardcoded tables are [initialised](https://github.com/TimothyStiles/poly/pull/350/files#diff-874dd545acc993503bc81ceab036740766e4192dbcb01dfdef9e1c91ec8400efR463), having an invalid table is not an error the caller can/will ever handle hence the panic. Should I refactor the initialisation code to handle the error? What would I do with the error apart from panicing?